### DISCUSSION TITLE
tutorial

### DISCUSSION THREAD

**Discussion by DavidSmolinski** [wwmm/easyeffects#1532] [Ideas]
I suggest making a tutorial to show how EasyEffects works.

**Comment by wwmm** [comment]
The idea is good. The problem is time. Most of the time I have to choose between fixing bugs / implementing features and improving the manual. And as there is just me and one more person working in the cod the manual haven't received much love. Help is welcome :-).

**Comment by p-chan5** [comment]
User rabcor wrote a [friendly tutorial](https://forum.manjaro.org/t/howto-enhance-your-linux-audio-with-easyeffects-or-pulseeffects-legacy/82497) on the manjaro forums. Maybe I can adapt it and incorporate it into the EE manual. How about calling it First Steps? By adapting it, I mean updating the Screenshots and tidying up the description of each element a bit.

Although I'm afraid I don't know `yelp` or basically how to show it on the technical side. But I could send you my sources and, whenever you have some time, add them to the package. What do you think?

**Reply by wwmm** [reply]
>How about calling it First Steps? 

I think it is fine. At least I do not have a better title now.

It is a good tutorial. But I would avoid the sections where commands like `pamac` are used. This will confuse people that use other distributions.

>Although I'm afraid I don't know yelp or basically how to show it on the technical side.

What we use to write the help is http://projectmallard.org/. It just happens yelp has good support for it. At the end of the day it is xml. You can take https://github.com/wwmm/easyeffects/blob/master/help/C/guide_1.page as example.  Just create a new file `guide_2.page` based on it.

**Reply by wwmm** [reply]
Although nowadays our manual does not have images they can be put there. Just take a look at our old PulseEffects files like these ones

https://github.com/wwmm/easyeffects/blob/pulseaudio-legacy/help/C/advancedinfo.page
https://github.com/wwmm/easyeffects/blob/pulseaudio-legacy/help/meson.build

**Reply by wwmm** [reply]
>What we use to write the help is http://projectmallard.org/. 

When I think about issues like #2046 I wonder if we should not try to build pdf instead. It would probably work in any desktop. But ideally I would like not having latex as build dependency. So another way to generate the pdf should be found if we follow this route.

**Reply by p-chan5** [reply]
> I would avoid the sections where commands like `pamac` are used. This will confuse people that use other distributions.

Yes, don't worry, it's meant for general UI-related purposes. I'll skip the installation part and things that have to do with the terminal.

> You can take https://github.com/wwmm/easyeffects/blob/master/help/C/guide_1.page as example. Just create a new file guide_2.page based on it.

Thanks for the suggestion. I'll make a rough draft using LibreOffice and export to PDF so you can suggest changes. I would like to have it ready before migrating to this format.

> Although nowadays our manual does not have images they can be put there. Just take a look at our old PulseEffects files like these ones

Yes, I noticed this in PE, I'll try to describe it as: the first look, at least from a novice user's point of view.

> When I think about issues like https://github.com/wwmm/easyeffects/issues/2046 I wonder if we should not try to build pdf instead. It would probably work in any desktop. But ideally I would like not having latex as build dependency. So another way to generate the pdf should be found if we follow this route.

Some projects use `.html` instead:

[GIMP Help](https://github.com/GNOME/gimp-help)
[LibreOffice Help](https://github.com/LibreOffice/help)

This would open the default browser, regardless of the user's desktop. Having a PDF is also a good idea, but I am concerned about how to edit it again, when thinking about new sections or corrections.

**Comment by wwmm** [comment]
>Some projects use .html instead:

At least on my computer GIMP says "there is no manual for my language installed" and redirects to a website. In any case one of the reasons for choosing mallard + yelp was translations. In the past we had gave up on the manual translation. But there is always someone who wants at least the ability to try to translate it. I am not sure about how to handle translations when using html.

**Comment by p-chan5** [comment]
This is my first draft. What do you think? If you agree with the wording then I should start migrating it to `xml` I may need some help with this, but I trust it will work out.

[easy_effects_manual_firts_steps.pdf](https://github.com/wwmm/easyeffects/files/10886417/easy_effects_manual_firts_steps.pdf)

I would like to work more in manual. Maybe I can do a tutorial on how to set the equalizer, although I'd need some time to think about how to approach it without confusing the reader... whatever! I'll go a little slow. We'll see what happens in the future.

> At least on my computer GIMP says "there is no manual for my language installed" and redirects to a website.

Yes, they must be installed manually. The package name starts with `gimp-help-` the files are located in `/usr/share/gimp/2.0/help`

> In any case one of the reasons for choosing mallard + yelp was translations. In the past we had gave up on the manual translation. But there is always someone who wants at least the ability to try to translate it. I am not sure about how to handle translations when using html.

I understand. Yes, we must consider the advantages of `yelp` Let's think of another way to do it.

**Reply by wwmm** [reply]
>This is my first draft. What do you think? If you agree with the wording then I should start migrating it to xml I may need some help with this, but I trust it will work out.

Some images resolution are a little low but the document is good :-). You can start the migration to the mallard xml.

>I'll go a little slow. We'll see what happens in the future

Definitely :-)

>Yes, they must be installed manually. The package name starts with gimp-help- the files are located in /usr/share/gimp/2.0/help

Oh...

>I understand. Yes, we must consider the advantages of yelp Let's think of another way to do it.

Meson has a "gnome module" that is making the `doc + translation magic` for yelp. Maybe there is a way to make it do the same for KDE...

**Reply by p-chan5** [reply]
> Some images resolution are a little low but the document is good :-).

Yes, its quality goes down when I export to PDF, it doesn't matter I still have the `.png` I also think I should remove the transparency of some... the shadows look bad. I will comment again when I have the job done. Until then!

**Comment by p-chan5** [comment]
I just remembered I left this here... 😧️

I'm really sorry I didn't finish it, and even if I wanted to, I no longer have enough time to pick it up again. 😢️

If anyone is interested in continuing it, please feel free to use my writing, although you might need to adapt it to the Qt version.
